MIPs/mip-x .md

```diff
@@ -0,0 +1,70 @@
+---
+MIP: X
+Title: Fix Error Handling in API Responses
+Status: Draft
+Stability: Experimental
+discussions-to: https://github.com/MetaMask/metamask-improvement-proposals/discussions
+Author(s): Shane Jonas (@shanejonas)
+Type: Maintaner
+Created: 2024-03-04
+---
+
+#### Abstract
+This proposal aims to refine MetaMask's API error handling by eliminating the practice of stringifying or wrapping errors. The current method obscures underlying issues, complicating debugging and resolution for developers.
+
+#### Motivation
+Issue https://github.com/MetaMask/metamask-extension/issues/19697 and https://github.com/MetaMask/metamask-extension/pull/15205#issuecomment-1199953855 highlighted difficulties faced by developers due to the API's error handling mechanism, where errors were stringified/wrapped, masking the root cause. This change proposes to break some existing error cases to facilitate clearer understanding and quicker resolution of issues.
```

Comment:
the errors are both wrapped and stringified, right? would it be more accurate to say that?